src/datepicker/datepicker.js

```diff
@@ -96,21 +103,25 @@ export default class Datepicker extends React.Component<
     this.props.onChange && this.props.onChange(data);
   };
 
-  formatDate(date: ?Date | Array<Date>, formatString: string) {
+  formatDate(date: ?T | Array<T>, formatString: string) {
+    const format = date => {
+      if (formatString === DEFAULT_DATE_FORMAT) {
```

Comment:
`date-io` has two different methods for formatting dates.  You can call `formatDate` which lets you provide a specific format string like `yyyy/mm/dddd`, or you can call `format` and pass in a named format string like `slashDate` or `fullDate` and then have that named format resolve to the right format string for the library being used in the date adapter. 

The tricky part here, is making those two scenarios work together correctly.  If `formatString` is provided as prop, I'm using formatDate, because its the responsibility of the user to make sure they pass the right date format for the library they're using, but if we're using the default, we need to make sure it's the correct default for the library.  

This is the case for both parsing and formatting